src/main/Member.js

```diff
@@ -784,12 +784,17 @@ export default class Member {
      * Cancels a token. (Called by the payer or the redeemer)
      *
      * @param {Token} token - token to cancel. Can also be a {string} tokenId
+     * @param {bool} blocking - creates a non-blocking request
      * @return {Promise} TokenOperationResult - cancelled token
      */
-    cancelToken(token) {
+    cancelToken(token, blocking) {
         return Util.callAsync(this.cancelToken, async () => {
             const finalToken = await this._resolveToken(token);
-            const cancelled = await this._client.cancelToken(finalToken);
+            const cancelled = await this._client.cancelToken(finalToken, blocking);
+            if (blocking && cancelled && cancelled.data &&
+              typeof cancelled.data.dispatchRequest === 'function') {
+              return cancelled.data.dispatchRequest;
```

Comment:
this method returns a function or a promise? can we make the interface uniform between sync and async calls so it's easier to use/understand? Perhaps return a function in both cases and have it wrap a promise?

Reply:
or perhaps split it into two functions since we need to special case it anyway?